### DISCUSSION TITLE
如何获取到mysql源数据引擎的表DDL

### DISCUSSION THREAD

**Discussion by BiuBiu2323** [juicedata/juicefs#5382] [Q&A]
juicefs大佬你们好，我因为一些限制无法使用root作为juicefs的用户，我希望使用一些普通的mysql用户，但我需要提前创建出juicefs所需要的表，可是我找不到DDL。请问我在那里可以获得meta的DDL呢？

**Comment by jiefenghuang** [comment]
授予指定db权限就可以了?
表定义在pkg/meta/sql.go:syncAllTables

**Comment by BiuBiu2323** [comment]
> 授予指定db权限就可以了? 表定义在pkg/meta/sql.go:syncAllTables

我给jucefs使用的用户没有建表的权限，我需要得到DDL，先创建出juicefs需要的表。请问有什么方式可以得到DDL sql吗

**Comment by jiefenghuang** [comment]
你用root跑一次创建好db和表, 然后在mysql中, "SHOW CREATE TABLE table_name;" 就好了吧
或者你在 syncAllTables 最后加个 "m.db.DumpAllToFile("schema.sql")", 跑一个sql load的单测, 比如TestLoadDump, 也可以输出全部建表sql.
